internal/handlers/pg/msg_saslstart.go

```diff
@@ -52,15 +53,23 @@ func (h *Handler) MsgSASLStart(ctx context.Context, msg *wire.OpMsg) (*wire.OpMs
 	}
 
 	parts := bytes.Split(payload.B, []byte{0})
-	if len(parts) != 3 || len(parts[0]) > 0 {
+	if l := len(parts); l != 3 {
 		return nil, common.NewCommandErrorMsgWithArgument(
 			common.ErrTypeMismatch,
-			"Invalid payload",
+			fmt.Sprintf("Invalid payload (expected 3 parts, got %d)", l),
 			"payload",
 		)
 	}
 
-	conninfo.Get(ctx).SetAuth(string(parts[1]), string(parts[2]))
+	authzid, authcid, passwd := parts[0], parts[1], parts[2]
+
+	// Some drivers (Go) send empty authorization identity (authzid),
+	// while others (Java) set it to the same value as authentication identity (authcid)
+	// (see https://www.rfc-editor.org/rfc/rfc4616.html).
+	// Ignore authzid for now.
+	_ = authzid
```

Comment:
So we assume/we are sure that we are still able to do authorization on every driver with just authcid, right?

Reply:
I don't think `authzid` is actually used by MongoDB for authorization. There is nothing in MongoDB docs that suggest that authentication and authorization identities could be different

Reply:
Yeah that wouldn't make sense